testnet/bitcoin-neon-controller/src/main.rs

```diff
@@ -170,12 +180,39 @@ async fn accept(addr: String, stream: TcpStream, config: &ConfigFile) -> http_ty
     Ok(())
 }
 
-async fn is_bootstrap_chain_required(config: &ConfigFile) -> http_types::Result<bool> {
+async fn is_chain_bootstrap_required(config: &ConfigFile) -> http_types::Result<bool> {
 
     let req = RPCRequest::is_chain_bootstrapped();
-    let stream = TcpStream::connect(config.neon.bitcoind_rpc_host.clone()).await?;
-    let body = serde_json::to_vec(&req).unwrap();
-    let mut resp = client::connect(stream.clone(), build_request(&config, body)).await?;
+
+    let mut backoff: f64 = 1.0;
+    let mut rng = thread_rng();
+    let mut resp =  loop {
+
+        backoff = (2.0 * backoff + (backoff * rng.gen_range(0.0, 1.0))).min(60.0);
+        let duration = Duration::from_millis((backoff * 1_000.0) as u64);
+
+        let stream = match TcpStream::connect(config.neon.bitcoind_rpc_host.clone()).await {
+            Ok(stream) => stream,
+            Err(e) => {
+                println!("Error while trying to connect to {}: {:?}", config.neon.bitcoind_rpc_host, e);
+                sleep(duration);
+                continue
+            }
+        };
+
+        let body = serde_json::to_vec(&req).unwrap();
+        let response = client::connect(stream.clone(), build_request(&config, body)).await;
```

Comment:
is the `stream.clone()` necessary?

Reply:
Indeed, addressed with https://github.com/blockstack/stacks-blockchain/pull/1645/commits/4d468878700957c68ebbcd22b4e66683105fbf75